realm/src/main/java/io/realm/internal/async/QueryUpdateTask.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal.async;
+
+import android.os.Handler;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.IdentityHashMap;
+import java.util.List;
+
+import io.realm.RealmConfiguration;
+import io.realm.RealmObject;
+import io.realm.RealmQuery;
+import io.realm.RealmResults;
+import io.realm.internal.SharedGroup;
+import io.realm.internal.Table;
+import io.realm.internal.TableQuery;
+import io.realm.internal.log.RealmLog;
+
+/**
+ * Manage the update of async queries.
+ */
+public class QueryUpdateTask implements Runnable {
+    // true if updating RealmResults, false if updating RealmObject, can't mix both
+    // the builder pattern will prevent this.
+    private boolean isUpdatingRealmResults;
```

Comment:
I think the confusion may occur with the name, not the fact that it's a boolean or enum. + using enum systematically to replace boolean is a bit extreme 😓